core/types/transaction.go

```diff
@@ -372,16 +372,23 @@ func (tx *Transaction) Hash() common.Hash {
 	return h
 }
 
-// Size returns the true RLP encoded storage size of the transaction, either by
-// encoding and returning it, or returning a previously cached value.
-func (tx *Transaction) Size() common.StorageSize {
+// Size returns the true encoded storage size of the transaction, either by encoding
+// and returning it, or returning a previously cached value.
+func (tx *Transaction) Size() uint64 {
 	if size := tx.size.Load(); size != nil {
-		return size.(common.StorageSize)
+		if tx.Type() == LegacyTxType {
+			return size.(uint64)
+		}
+		return size.(uint64) + 1 // type byte
 	}
 	c := writeCounter(0)
 	rlp.Encode(&c, &tx.inner)
-	tx.size.Store(common.StorageSize(c))
-	return common.StorageSize(c)
+	tx.size.Store(uint64(c))
```

Comment:
If you're doing this, then I suggest renaming it to `tx.rlpSize`, so it's clear that it doesn't cache `Size`

Reply:
Reverted it to be the real `size` per your PR.